File: Leetcode Daily Challenge/027-September-2025/007-find-n-unique-integers-sum-up-to-zero.cpp
// https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/




#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> sumZero(int n) {
        vector<int> result;
        for (int i = 1; i <= n/2; i ++) {
            result.push_back(i);
            result.push_back(-i);
        }
        if (n % 2) result.push_back(0);
        return result;
    }
};

File: Leetcode Daily Challenge/027-September-2025/007-find-n-unique-integers-sum-up-to-zero.java
// https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/




class Solution {
    public int[] sumZero(int n) {
        int[] result = new int[n];
        for (int i = 1; i <= n / 2; i ++) {
            result[i - 1] = i;
            result[n - i] = -i;
        }
        return result;
    }
}